pkg/controller/controller_multiplex.go

```diff
@@ -0,0 +1,175 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controller
+
+import (
+	"fmt"
+
+	"github.com/golang/glog"
+
+	"github.com/kubernetes-incubator/service-catalog/pkg/apis/servicecatalog/v1beta1"
+	"github.com/kubernetes-incubator/service-catalog/pkg/pretty"
+	corev1 "k8s.io/api/core/v1"
+
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+// Conditions
+
+// ConditionType can be translated to a ServiceInstanceCondition or
+// ServiceBindingCondition value.
+type ConditionType string
```

Comment:
Shouldn't we be using the ones defined in types vs redefining our own?

Reply:
I did this because there are no generic ConditionType, the ConditionTypes are typed but it is not something that will ever be saved into the object, it is more for use in the generic operation as seen in this file